crypto/mem_clr.c

```diff
@@ -60,6 +60,7 @@
 #include <string.h>
 #include <openssl/crypto.h>
 
+extern unsigned char cleanse_ctr;
```

Comment:
I've seen this sort of warning with a sufficiently recent clang and sufficiently aggressive warning settings.  I'm not sure if 3.6 is a new enough clang or 3.7 would be needed, though; the Apple Xcode bits seem to no longer say what LLVM version they're based off of.